Made TorrentDef.load async and threaded

qstokkink opened the pull request:
Related to #5399 (PR 2/4)

This PR moves `TorrentDef.load` to a thread. The fallout from this - essentially 12-line - change consists of 245 additions and 309 deletions across 21 files.

Our codecov check uncovered some dead code, which I removed (instead of writing tests for it):
- `tribler/core/components/metadata_store/tests/gen_test_data.py`
- `tribler/core/components/metadata_store/restapi/tests/test_channels_endpoint.py (test_add_torrent_from_magnet)`

kozlovsky reviewed (changes requested):
Overall looks good, some bugs need to be addressed

qstokkink commented:
@kozlovsky Thanks for your review. I addressed your points pretty much one-to-one according to your suggestions. Regarding `start_handle`, I adopted your suggestion of returning `None`.

I requested your re-review: please ensure that I indeed adopted your suggestions as you intended.

kozlovsky reviewed (approved):
Everything looks good now!
